hw02/EnemyCollision.cs

```diff
@@ -53,4 +64,25 @@ void Shoot ()
 
     Remember to call your Coroutine.
     */ 
+    
+    public class Weapon: MonoBehaviour
+    {
```

Comment:
Awesome! The switch logic can also be before the yield return statement, that way the weapon will be switched first before the waiting. If you put it after, it will wait 5 seconds first.